iroha/src/block_sync.rs

```diff
@@ -368,20 +368,12 @@ pub mod message {
         /// Send this message over the network to the specified `peer`.
         #[iroha_futures::telemetry_future]
         #[log("TRACE")]
-        pub async fn send_to(self, peer: &PeerId) -> Result<()> {
-            let message: VersionedMessage = self.into();
-            match Network::send_request_to(
-                &peer.address,
-                Request::new(uri::BLOCK_SYNC_URI, message.encode_versioned()?),
-            )
-            .await?
-            {
-                Response::Ok(_) => Ok(()),
-                Response::InternalError => Err(error!(
-                    "Failed to send message - Internal Error on peer: {:?}",
-                    peer
-                )),
-            }
+        pub async fn send_to(self, broker: &Broker, peer: &PeerId) -> Result<()> {
+            let id = iroha_p2p::peer::PeerId::from(peer.clone());
```

Comment:
It is better to push as much allocations as possible to user. So you can pass here `PeerId` by value